### DISCUSSION TITLE
Repeating channels in every country's playlists

### DISCUSSION THREAD

**Discussion by arecki2** [iptv-org/community#294] [Q&A] [answered]
Hi there,

I found this repo by accident recently an I've been fiddling around with the playlists. Overall, an awesome work guys! 👍

I found one thing strange, however.

Usually, I would either load a playlist for a specific country or the one for all countries... and I see that some channels keep appearing under every country I select. Some appear regionally, such as CNBC Europe or AzerbaijanNewsTV.se (?!) under every European country, or BBC Earth with Vietnamese subtitles under every Asian one... but a lot more, for example Insight TV, Red Bull TV, TRT World, Al Jazeera, Euro Indie Music Chart, various Euronews versions, or - God forbid - ruSSia today - are present in every country's playlist. Regardless if it's UK, US, Japan, Poland, or freakin' Cayman Islands - they're all there. And there are many more - why on Earth should one need Berita Satu (from Indonesia) or Bangladeshi or Mongolian "world" channels (BTV World, MNB World, in fact any other channel with "world" in its name), or ever-non-working SIC International (that one is from Portugal) while looking for channels from e.g. Brazil?

Furthermore, these channels appear multiple times in index.country.m3u, so I get as many Berita Satu dupes as there are countries & small island territories out there. (Honestly, I never thought Isle of Man would deserve an own playlist, even though there are no TV channels genuinely based there listed in that playlist!)

Why is that? Perhaps there is a good reason for exposing these channels to the world like that... or is it a bug? Just wondering 🤔

**Comment by BellezaEmporium** [comment, marked as answer]
Hello,

Those are channels that are streamed internationally. (Of course, some countries like DPRK or China do control their streams and channels). This is why they are seen on every country.

**Reply by loucasal** [reply]
The underlying assumption is that most users are interested in a single country playlist. Is this backed by some evidence?
For any users who use more than one country playlist, a separate international category (or regional categories) would be much more convenient.

Leaving this to users to address manually downstream is also not an option, as links keep changing and playlist editors readd them as new channels.

**Reply by BellezaEmporium** [reply]
We used to have an international playlist before, though it was clear that it caused more confusion in the early days because there was programs that weren't getting correctly inside the proper playlists. We've then decided to put the international channels in their country of origin and broaden the channels to all countries. 

I can understand the fact that it feels weird for some persons to have 40+ channels that are irrelevant compared to the ones in their countries. The only option that's available on our side is to treat the channels that are streamed internationally as such (by changing their tvg-id as an example, going from .us to .int), but this would cause the EPG not to work, and it would cause a major rework of our playlist creation (and most certainly our database)

**Reply by BellezaEmporium** [reply]
Though I think we could add a tab inside our database to see if the channel is streamed worldwide or not, it would need a big knowledge of all international channels. I invoke @freearhey in this discussion to elaborate on this subject.

**Reply by freearhey** [reply]
It is already possible to mark a channel as international. To do this, just specify the region `r/INT` in the `broadcast_area` field. 
https://github.com/iptv-org/database/blob/master/CONTRIBUTING.md#channels

Full list of supported regions: https://github.com/iptv-org/iptv/blob/master/.readme/supported-regions.md

**Reply by BellezaEmporium** [reply]
> It is already possible to mark a channel as international. To do this, just specify the region `r/INT` in the `broadcast_area` field. https://github.com/iptv-org/database/blob/master/CONTRIBUTING.md#channels
> 
> Full list of supported regions: https://github.com/iptv-org/iptv/blob/master/.readme/supported-regions.md

The thing would be, in the same pace, to have International in both "Regions" and "Countries", to make sure that it's got its separate part. It'll prevent us from having the issue OP and @loucasal are bringing.

**Reply by BellezaEmporium** [reply]
I bump back this topic. Maybe we should make sure that all international channels get into a "int.m3u" file instead of getting sparsed all over the different playlists.

**Reply by Takakeisho** [reply]
It's really annonying, i tried to hide them on Kodi but they are back 2-3 days later. I understand if someone is using a single country playlist but for someone who using multiple playlist, it's a pain in the lower back.

**Comment by AntiPontifex** [comment]
My 2 cents: bring back the int.3u playlist.
